flink-streaming-java/src/main/java/org/apache/flink/streaming/api/environment/StreamContextEnvironment.java

```diff
@@ -75,29 +62,31 @@ public JobExecutionResult execute() throws Exception {
 	@Override
 	public JobExecutionResult execute(String jobName) throws Exception {
 
-		JobGraph jobGraph;
-		if (jobName == null) {
-			jobGraph = this.getStreamGraph().getJobGraph();
-		} else {
-			jobGraph = this.getStreamGraph().getJobGraph(jobName);
-		}
-
-		transformations.clear();
-
-		// attach all necessary jar files to the JobGraph
-		for (URL file : jars) {
```

Comment:
Jars and classpaths also need to be attached to the JobGraph of the non-detached streaming job, right? Wouldn't it be nice to always do that at the Client? It's always good to avoid multiple branches for doing the same thing. `Client.getJobGraph(FlinkPlan,..` enables you to pass the streamgraph directly with the jars and classpaths.